util/types/json/impl.go

```diff
@@ -57,8 +64,9 @@ func normalize(in interface{}) JSON {
 			array[i] = normalize(elem)
 		}
 		return jsonArray(array)
+	default:
+		panic("unsupported json type")
```

Comment:
should we support [jsonDecimal](https://github.com/mysql/mysql-server/blob/5.7/sql/json_dom.cc#L2941)?